Feat - ZRangePicker (#267)
ZanichelliEditore/design-system: src/components/date-picker/z-range-picker/index.stories.mdx

import { html } from "lit-html";
import { Meta, Canvas, Story, ArgsTable } from "@storybook/addon-docs";

import { ZDatePickerMode } from "../../../beans";

import "./index";
import "../styles.css";

<Meta
  title="DatePicker/ZRangePicker"
  component="z-range-picker"
  argTypes={{
    mode: {
      options: Object.values(ZDatePickerMode),
      control: { type: "radio" },
    },
    rangePickerId: {
      control: "text",
    },
  }}
/>

# ZDatePicker with date

<Canvas>
  <Story
    name="ZDatePicker with date"
    args={{
      mode: "date",
      rangePickerId: "picker-01",
    }}

>     height="180px"

Reviewer: non penso funzioni questo attributo

  >
    {(args) => html`
      <script>
        const picker1 = document.getElementById("${args.rangePickerId}");
        picker1.addEventListener("dateSelect", (e) => {
          const input = document.getElementById("output");
          input.innerHTML = e.detail;
        });
      </script>
      <div style="display:flex">
        <div style="height:300px; padding-top:40px">
          <z-range-picker
            mode=${args.mode}
            rangePickerId=${args.rangePickerId}
            id=${args.rangePickerId}
          ></z-range-picker>
        </div>
        <div

>           style="display:flex; flex-direction: column; margin-left: 100px; border:1px solid black; width:400px; margin-top:40px; height:80px"

Reviewer: visto che c'è tanto stile inline, sarebbe meglio fare un file css a parte `index.stories.css` e importarlo qui

import { html } from "lit-html";
import { Meta, Canvas, Story, ArgsTable } from "@storybook/addon-docs";

import { ZDatePickerMode } from "../../../beans";

import "./index";
import "../styles.css";

<Meta
  title="DatePicker/ZRangePicker"
  component="z-range-picker"
  argTypes={{
    mode: {
      options: Object.values(ZDatePickerMode),
      control: { type: "radio" },
    },
    rangepickerid: {
      control: "text",
    },
  }}
/>

# ZDatePicker with date

<Canvas>
  <Story
    name="ZDatePicker with date"
    args={{
      mode: "date",
      rangepickerid: "picker-01",
    }}
    height="180px"
  >
    {(args) => html`
      <div style="height:300px; padding-top:40px">
        <z-range-picker
          mode=${args.mode}
          rangepickerid=${args.rangepickerid}
          id=${args.rangepickerid}
        ></z-range-picker>
      </div>
      <script>
        const picker1 = document.getElementById("${args.rangepickerid}");
        picker1.addEventListener("dateSelect", (e) => console.log(e.detail));

>       </script>

Author: È un modo per mostrare che viene restituito il valore del range selezionato, lo sostituisco con un box a lato anziché un console.log

    `}
  </Story>
</Canvas>

# ZDatePicker with date and time

<Canvas>
  <Story
    name="ZDatePicker with date and time"
    args={{
      mode: "date-time",
      rangepickerid: "picker-02",
    }}
  >
    {(args) => html`
      <div style="height:300px; padding-top:40px">
        <z-range-picker
          mode=${args.mode}
          rangepickerid=${args.rangepickerid}
          id=${args.rangepickerid}
        ></z-range-picker>
      </div>
      <script>
        const picker2 = document.getElementById("${args.rangepickerid}");
        picker2.addEventListener("dateSelect", (e) => console.log(e.detail));
      </script>
    `}
  </Story>
</Canvas>

## Usage

In order to avoid conflits between datepickers, pass a unique id as `rangepickerid` prop.

> This component has a callback function `dateSelect` that returns the selected date.

Reviewer: ```suggestion
This component emits an event called `dateSelect` with the value of the selected date.
```
